#!/usr/bin/env python
PACKAGE = "rcbigctl"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kp_X",  double_t, 0, "Position Controller X Kp", 1.8, 0, 10)
gen.add("Max_X", double_t, 0, "Position Controller Max X Velocity", 1.0, 0, 10)
gen.add("A_X",   double_t, 0, "Position Controller X Accel", 2.0, 0, 10)

gen.add("Kp_Y", double_t, 0, "Position Controller Y Kp", 1.8, 0, 10)
gen.add("Max_Y", double_t, 0, "Position Controller Max Y Velocity", 1.0, 0, 10)
gen.add("A_Y", double_t, 0, "Position Controller Y Accel", 2.0, 0, 10)

gen.add("Kp_W", double_t, 0, "Position Controller W Kp", 1.0, 0, 10)
gen.add("Max_W", double_t, 0, "Position Controller Max W Velocity", 1.4, 0, 10)
gen.add("A_W", double_t, 0, "Position Controller W Accel", 2.5, 0, 10)

exit(gen.generate(PACKAGE, "controller_node", "Controller"))